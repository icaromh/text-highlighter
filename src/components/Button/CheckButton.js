import React, { Component } from 'react'
import PropTypes from 'prop-types'

import { SavedIcon } from '../Icons'

class CheckButton extends Component {
  constructor(props) {
    super(props)
    this.state = {
      checked: false,
    }

    this.handleOnClick = this.handleOnClick.bind(this)
  }

  handleOnClick() {
    const { value, onClick } = this.props

    this.setState(state => ({
      checked: !state.checked,
    }), () => onClick(value, this.state.checked)) // eslint-disable-line react/destructuring-assignment
  }

  render() {
    const { value } = this.props
    const { checked } = this.state
    const checkedClass = checked ? 'checked' : ''

    return (
      <button
        type="button"
        className={`Button ${checkedClass}`}
        onClick={this.handleOnClick}
      >
        <SavedIcon color={value} />
      </button>
    )
  }
}

CheckButton.propTypes = {
  value: PropTypes.string.isRequired,
  onClick: PropTypes.func.isRequired,
}

export default CheckButton
