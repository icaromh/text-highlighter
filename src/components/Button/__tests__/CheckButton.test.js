import React from 'react'
import { shallow } from 'enzyme'

import CheckButton from '../CheckButton'

describe('#CheckButton', () => {
  test('expect to render', () => {
    const wrapper = shallow(<CheckButton value="red" onClick={() => {}} />)
    expect(wrapper).toMatchSnapshot()
  })

  test('expect to perform onClick n update class', () => {
    const onClickMock = jest.fn()
    const wrapper = shallow(<CheckButton value="red" onClick={onClickMock} />)

    wrapper.simulate('click')
    expect(onClickMock).toHaveBeenCalled()
    expect(wrapper.find('.Button').hasClass('checked')).toBe(true)

    wrapper.simulate('click')
    expect(onClickMock).toHaveBeenCalled()
    expect(wrapper.find('.Button').hasClass('checked')).toBe(false)
  })
})
