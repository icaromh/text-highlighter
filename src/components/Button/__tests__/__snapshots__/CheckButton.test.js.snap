// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#CheckButton expect to render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CheckButton
    onClick={[Function]}
    value="red"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <SavedIcon
        color="red"
      />,
      "className": "Button ",
      "onClick": [Function],
      "type": "button",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "color": "red",
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "button",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <SavedIcon
          color="red"
        />,
        "className": "Button ",
        "onClick": [Function],
        "type": "button",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "red",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "button",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
